sync/sync.go

```diff
@@ -311,30 +311,44 @@ Register:
 	return t.reloadAndVerify(s, replicaAddress, toClient)
 }
 
-func (t *Task) isRevisionCountAndChainSame(fromClient, toClient *replicaClient.ReplicaClient) (bool, error) {
+func (t *Task) isRevisionCountAndChainSame(fromClient, toClient *replicaClient.ReplicaClient) (bool, []string, error) {
 	rwReplica, err := fromClient.GetReplica()
 	if err != nil {
-		return false, err
+		return false, nil, err
 	}
 
 	curReplica, err := toClient.GetReplica()
 	if err != nil {
-		return false, err
+		return false, nil, err
 	}
 	logrus.Infof("RevisionCount of RW replica (%v): %v, Current replica (%v): %v",
 		fromClient.GetAddress(), rwReplica.RevisionCounter, toClient.GetAddress(),
 		curReplica.RevisionCounter)
 	logrus.Infof("RW replica chain: %v, cur replica chain: %v", rwReplica.Chain, curReplica.Chain)
+	if curReplica.Checkpoint != "" {
```

Comment:
lets add the comments here why there is no need to check for below the checkpoint?

Reply:
if I'm not wrong, replica should confirm on checkpoint before proceeding for deletion?

Reply:
Yes replica verifies the checkpoint from controller before proceeding for snapshot deletion. This will be pushed in proceeding PR.